flight/Modules/VibrationAnalysis/vibrationanalysis.c

```diff
@@ -83,17 +84,54 @@ static struct VibrationAnalysis_data {
 	float accels_static_bias_y; // (0,0,-g). In the case where they are not, this will still  
 	float accels_static_bias_z; // converge to the true bias in a few thousand measurements.	
 	
-	int16_t *accel_buffer_complex_x_q15;
-	int16_t *accel_buffer_complex_y_q15;
-	int16_t *accel_buffer_complex_z_q15;
-	
-	int16_t *fft_output;
+	int16_t *accel_buffer_x;
+	int16_t *accel_buffer_y;
+	int16_t *accel_buffer_z;
 } *vtd;
 
 
 // Private functions
 static void VibrationAnalysisTask(void *parameters);
 
+/*
+*   Releases any memory dinamically allocated
+*   Because this module can have a big footprint this will ensure we can
+*   reduce it when something fails.
```

Comment:
But in this case every time settings change a new buffer is allocated.

The options I see:
- Pre-allocate the maximum size (1024)
- Disable dynamic allocation: will imply a reboot when changing values
- Try to use the stack only: probably will present issues with larger number of sample buffers
- Implement PIOS_free